packages/components/src/navigator/types.ts

```diff
@@ -12,16 +12,27 @@ export type NavigatorLocation = NavigateOptions & {
 	isBack?: boolean;
 	path?: string;
 	hasRestoredFocus?: boolean;
+	restoreFocusTo?: string;
 };
 
 export type NavigatorContext = {
 	location: NavigatorLocation;
+	params: any;
+	match?: string;
 	goTo: ( path: string, options?: NavigateOptions ) => void;
 	goBack: () => void;
+	addScreen: ( screen: NavigatorScreen ) => void;
+	removeScreen: ( screen: NavigatorScreen ) => void;
 };
 
 // Returned by the `useNavigator` hook.
-export type Navigator = NavigatorContext;
+export type Navigator = {
```

Comment:
This is more like the "public type". While the one with "addScreen and removeScreen" is the private one.

Reply:
Makes sense. I've suggested an improvement in https://github.com/WordPress/gutenberg/pull/47827#discussion_r1098824628